docs/design.md

```diff
@@ -0,0 +1,785 @@
+* `metadata.label` - string, optional, repeatable - Uses the format
+  `metadata.label=key=value`. Repeating this parameter indicates all
+  requested labels must apply (AND not OR).
+
+### GET `/revisions/{revision_id}/rendered-documents`
+
+Returns a multi-document YAML of fully layered and substituted documents. No
+abstract documents will be returned. This is the primary endpoint that
+consumers will interact with for their configuration.
+
+Valid query parameters are the same as for
+`/revisions/{revision_id}/documents`, minus the paremters in
+`metadata.layeringDetinition`, which are not supported.
+
+### GET `/revisions`
+
+Lists existing revisions and reports basic details including a summary of
+validation status for each `deckhand/ValidationPolicy` that is part of that
+revision.
+
+Sample response:
+
+```yaml
+---
+count: 7
+next: https://deckhand/revisions?limit=2&offset=2
+prev: null
+results:
+  - id: 0
+    url: https://deckhand/revisions/0
+    createdAt: 2017-07-14T21:23Z
+    validationPolicies:
+      site-deploy-validation:
+        status: failed
+  - id: 1
+    url: https://deckhand/revisions/1
+    createdAt: 2017-07-16T01:15Z
+    validationPolicies:
+      site-deploy-validation:
+        status: succeeded
+...
+```
+
+### GET `/revisions/{{revision_id}}`
+
+Get a detailed description of a particular revision.
+
+Sample response:
+
+```yaml
+---
+id: 0
+url: https://deckhand/revisions/0
+createdAt: 2017-07-14T021:23Z
+validationPolicies:
+  site-deploy-validation:
+    url: https://deckhand/revisions/0/documents?schema=deckhand/ValidationPolicy/v1&name=site-deploy-validation
+    status: failed
+    validations:
+      - name: deckhand-schema-validation
+        url: https://deckhand/revisions/0/validations/deckhand-schema-validation/0
+        status: success
+      - name: drydock-site-validation
+        status: missing
+      - name: promenade-site-validation
+        url: https://deckhand/revisions/0/validations/promenade-site-validation/0
+        status: expired
+      - name: armada-deployability-validation
+        url: https://deckhand/revisions/0/validations/armada-deployability-validation/0
+        status: failed
+...
+```
+
+Validation status is always for the most recent entry for a given validation.
+A status of `missing` indicates that no entries have been created. A status
+of `expired` indicates that the validation had succeeded, but the
+`expiresAfter` limit specified in the `ValidationPolicy` has been exceeded.
+
+### POST `/revisions/{{revision_id}}/validations/{{name}}`
+
+Add the results of a validation for a particular revision.
+
+An example `POST` request body indicating validation success:
+
+```yaml
+---
+status: succeeded
+validator:
+  name: promenade
+  version: 1.1.2
+...
+```
+
+An example `POST` request indicating validation failure:
+
+```http
+POST /revisions/3/validations/promenade-site-validation
+Content-Type: application/x-yaml
+
+---
+status: failed
+errors:
+  - documents:
```

Comment:
The schema format below doesn't appear in the detailed description for a revision (under validations section), i.e. for "GET `/revisions/{{revision_id}}`" above. Shouldn't it?